frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/wrappers.tsx

```diff
@@ -1,25 +1,39 @@
 import { useCallback, useEffect, useState } from "react";
-import { push } from "react-router-redux";
+import { push, replace } from "react-router-redux";
 
-import { useDispatch } from "metabase/lib/redux";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { getLocation } from "metabase/selectors/routing";
 
 import { TablePicker } from "./TablePicker";
-import type { TreePath } from "./types";
+import type { ChangeOptions, TreePath } from "./types";
 import { getUrl } from "./utils";
 
 export function RouterTablePicker(props: TreePath) {
   const dispatch = useDispatch();
   const [value, setValue] = useState(props);
+  const location = useSelector(getLocation);
+  const isSegments = location.pathname?.startsWith("/admin/datamodel/segment");
+
   const onChange = useCallback(
-    (value: TreePath) => {
+    (value: TreePath, options?: ChangeOptions) => {
       setValue(value);
-      dispatch(push(getUrl(value)));
+
+      // prevent auto-navigation from table-picker when Segments tab is open
+      if (!isSegments) {
+        if (options?.isAutomatic) {
+          dispatch(replace(getUrl(value)));
```

Comment:
There's a test (which makes sense) that every automatic redirect should use `replace`.

Alternatively I was thinking to leave `onChange` as is, but add one more prop to `TablePicker`, e.g. `onRedirect` or something.